File: archives/20190519python/0350. Intersection of Two Arrays II.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        dict_num={}
        for num in nums1:
            if num not in dict_num:
                dict_num[num]=1
            else:
                dict_num[num]+=1
        rep=[]
        for num in nums2:
            if num in dict_num and dict_num[num]>0:
                rep.append(num)
                dict_num[num]-=1
        return rep

File: archives/C#/0350. Intersection of Two Arrays II.cs
public class Solution {
    public int[] Intersect(int[] nums1, int[] nums2) {
        IList<int> rep=new List<int>();
        Dictionary<int,int> numsDict=new Dictionary<int,int>();
        foreach(int num in nums1){
            if(!numsDict.ContainsKey(num)){
                numsDict[num]=1;
            }
            else{
                numsDict[num]+=1;
            }
        }
        foreach(int num in nums2){
            if(numsDict.ContainsKey(num) && numsDict[num]>0){
                numsDict[num]-=1;
                rep.Add(num);
            }
        }
        return rep.ToArray();
    }
}
